src/window/update_loop.rs

```diff
@@ -325,6 +350,7 @@ impl ApplicationHandler<UserEvent> for UpdateLoop {
         match event {
             WindowEvent::RedrawRequested => {
                 self.redraw_requested();
+                self.maximize_x11();
```

Comment:
Not sure if this would be a good place to trigger the check, please let me know if you think it should be moved somewhere else.